src/common/featuremanager/FeatureManagerService.js

```diff
@@ -956,6 +956,18 @@
       }
       return deferredResponse.promise;
     };
+
+    this.coordinatesWithinRange = function(coords1, coords2, range) {
+      if (coords1.length != coords2.length) {
+        return false;
+      }
+      for (var i = 0; i < coords1.length; i++) {
```

Comment:
Javascript thing, `for (var i = 0, ii = coords1.length; i < ii; i++) {`, will be substantially faster than calling ".length"  repetitively.